### ISSUE TITLE
Finish docker-compose config

### ISSUE THREAD

**Issue by MichaelKohler** [mozillacommunitych/mozilla.ch#91]
WIP Branch: https://github.com/mozillach/mozilla.ch/tree/feature/docker_compose

- [ ] Add watcher for changes (rebuilding css etc) if run for development
- [ ] Make sure that the build is still valid and can be built on DockerHub
- [ ] Document how it needs to be run for production
- [ ] Tell Community IT to get to this new architecture (@MichaelKohler)
- [ ] Nuke Vargrant

**Comment by freaktechnik** [comment]
- [ ] Nuke Vargrant?

**Comment by freaktechnik** [comment]
Not sure community IT can run docker compose, but honestly never looked into mesos that deeply.

**Comment by MichaelKohler** [comment]
True, but the normal Dockerfile + arguments should still work, right?

**Comment by freaktechnik** [comment]
> True, but the normal Dockerfile + arguments should still work, right?

With the current production compose settings for sure (it's just three env vars and a port forward).

**Comment by MichaelKohler** [comment]
Good. Let's see if somebody has time to do it so it works with docker-compose and only the normal Dockerfile :)

**Comment by bfirsh** [comment]
(Compose author here. 👋 )

Generally looks good! A few little things:

* I prefer having the default `docker-compose.yml` to be the dev environment, then having `docker-compose.prod.yml` or whatever for production. That way in development, you can just type `docker-compose up` instead of remembering a long command.

  The downside of this is you can't get rid of volumes when inheriting, so you have to write out the prod file in full. (Yeah, this sucks, but the `extends` option is coming back soon I think which will make this better. On the plus side, it means you have a nice explicit production file instead of worrying about what gets overridden where.)
* You can put environment variables in a `.env` file instead of putting them on the command-line every time. In this case, `.env` would contain `MOZILLIANS_KEY={mozillians API key}`
* I like pointing users directly at Docker for Mac instead of the Compose installation instructions because that is normally what they want. ([Here's an example that I wrote.](https://github.com/embassynetwork/modernomad/blob/master/docs/docker-development-environment.md))

**Comment by freaktechnik** [comment]
> I prefer having the default docker-compose.yml to be the dev environment, then having docker-compose.prod.yml or whatever for production. That way in development, you can just type docker-compose up instead of remembering a long command.

Given that the main application will probably end up being development this would make a lot of sense. 

Thanks for your input!
